metrics/isolator.cpp

```diff
@@ -0,0 +1,379 @@
+            request.keepAlive = true;
+            request.headers = {
+              {"Accept", APPLICATION_JSON},
+              {"Content-Type", APPLICATION_JSON}};
+
+            if (body.isSome()) {
+              request.body = body.get();
+            }
+
+            if (serviceInetAddress.isSome()) {
+              request.url = http::URL(
+                  serviceScheme,
+                  serviceInetAddress->ip,
+                  serviceInetAddress->port,
+                  endpoint);
+            }
+
+            if (serviceUnixAddress.isSome()) {
+              request.url.scheme = serviceScheme;
+              request.url.domain = "";
+              request.url.path = endpoint;
+            }
+
+            return connection.send(request);
+          }))
+      .after(
+          flags.request_timeout,
+          [](const Future<http::Response>&) {
+            return Failure("Request timed out");
+          });
+  }
+
+  Future<http::Response> send(const ContainerStartRequest& containerStartRequest)
+  {
+    string body = jsonify(JSON::Protobuf(containerStartRequest));
+
+    return send(serviceEndpoint, body, "POST");
+  }
+
+  Future<http::Response> sendContainerStop(const ContainerID& containerId)
+  {
+    return send(serviceEndpoint + "/" + containerId.value(), None(), "DELETE");
+  }
+
+private:
+  const isolator::Flags flags;
+  string serviceScheme;
+  string serviceEndpoint;
+  Option<inet::Address> serviceInetAddress;
+  Option<unix::Address> serviceUnixAddress;
+};
+
+
+MetricsIsolator::MetricsIsolator(const isolator::Flags& flags)
+  : process(new MetricsIsolatorProcess(flags))
+{
+  spawn(process.get());
+}
+
+
+MetricsIsolator::~MetricsIsolator()
+{
+  terminate(process.get());
+  wait(process.get());
+}
+
+
+Future<Option<ContainerLaunchInfo>> MetricsIsolator::prepare(
+    const ContainerID& containerId,
+    const slave::ContainerConfig& containerConfig)
+{
+  return dispatch(process.get(),
+                  &MetricsIsolatorProcess::prepare,
+                  containerId,
+                  containerConfig);
+}
+
+
+Future<Nothing> MetricsIsolator::cleanup(
+    const ContainerID& containerId)
+{
+  return dispatch(process.get(),
+                  &MetricsIsolatorProcess::cleanup,
+                  containerId);
+}
+
+} // namespace metrics {
+} // namespace dcos {
+} // namespace mesosphere {
+
+mesos::modules::Module<Isolator>
+com_mesosphere_dcos_MetricsIsolatorModule(
+    MESOS_MODULE_API_VERSION,
+    MESOS_VERSION,
+    "Mesosphere",
+    "support@mesosphere.com",
+    "Metrics Isolator Module.",
+    nullptr,
+    [](const Parameters& parameters) -> Isolator* {
+      // Convert `parameters` into a map.
+      map<string, string> values;
+      foreach (const Parameter& parameter, parameters.parameter()) {
+        values[parameter.key()] = parameter.value();
+      }
+
+      mesosphere::dcos::metrics::isolator::Flags flags;
+
+      // Load and validate flags from environment and map.
```

Comment:
Looks like you forgot to update this comment when removing the flags-parsing-from-environment-variables.